simplefs/simplefs.go

```diff
@@ -950,15 +1173,42 @@ func (k *SimpleFS) SimpleFSCancel(_ context.Context, opid keybase1.OpID) error {
 // SimpleFSCheck - Check progress of pending operation
 // Progress variable is still TBD.
 // Return errNoResult if no operation found.
-func (k *SimpleFS) SimpleFSCheck(_ context.Context, opid keybase1.OpID) (keybase1.Progress, error) {
+func (k *SimpleFS) SimpleFSCheck(_ context.Context, opid keybase1.OpID) (keybase1.OpProgress, error) {
 	k.lock.RLock()
 	defer k.lock.RUnlock()
-	if _, ok := k.inProgress[opid]; ok {
-		return 0, nil
+	if p, ok := k.inProgress[opid]; ok {
+		// For now, estimate the ending time purely on the read progress.
+		var n, d int64
+		progress := p.progress
+		if progress.BytesTotal > 0 {
+			n = progress.BytesRead
+			d = progress.BytesTotal
+		} else if p.progress.FilesTotal > 0 {
+			n = progress.FilesRead
+			d = progress.FilesTotal
+		}
+		if n > 0 && d > 0 && !progress.Start.IsZero() &&
+			progress.EndEstimate.IsZero() {
```

Comment:
Perhaps we should recompute `EndEstimate` on every `SimpleFSCheck` call instead of only first time?

Reply:
@songgao: we do recompute it every time, until the operation ends.  `progress` is a copy of `p.progress`, which isn't a pointer, but just a regular `keybase1.OpProgress`.

Reply:
I see; makes sense. It must be the bug I had earlier in my local branch that made me think this was not recomputed.

Why the `progress.EndEstimate.IsZero()` check though?

Reply:
That's checking if the operation has ended, and if `EndEstimate` was set in the actual map.

Reply:
Oh I see this is for when after `doneOp` is called. NVM. Thanks!

Reply:
Ugh! GitHub again failed to show me the new comment before I submitted my comment.